File: duckerp/duckerp/utils/quotation.js
frappe.ui.form.on("Quotation", {
    custom_fetch_items: function(frm) {
        if(frm.doc.custom_group_no){
            frappe.call({
                method: "duckerp.duckerp.utils.quotation.quotation_item_adding",
                args: { custom_group_no: frm.doc.custom_group_no, customer: frm.doc.party_name },
                callback: function(r) {
                    if (r.message) {
                        r.message.forEach((item) => {
                            let child = frm.add_child("items");
                            frappe.model.set_value(child.doctype, child.name, "item_code", item.item);
                            frappe.model.set_value(child.doctype, child.name, "custom_group_no", item.name);
                            frappe.model.set_value(child.doctype, child.name, "custom_group_name", item.part_group_name);
                        });
                        frm.refresh_field("items");
                    }
                }
            });
        }
    },
    party_name: function(frm){
       frm.clear_table("items");
    }
});

File: duckerp/duckerp/utils/quotation.py
import frappe

@frappe.whitelist()
def quotation_item_adding(custom_group_no, customer):
    if custom_group_no and customer:
        part_no=frappe.db.get_value("Part Group", {"name":custom_group_no}, "part_group_name")
        if part_no:
            part_groups=frappe.db.get_all("Part Group", filters={"part_group_name":part_no, "customer_name":customer}, fields=["item", "name", "part_group_name"])
            return part_groups